src/silo/config/preprocessing_config.cpp

```diff
@@ -57,25 +57,22 @@ std::optional<std::filesystem::path> PreprocessingConfig::getNdjsonInputFilename
              : std::nullopt;
 }
 
-std::filesystem::path PreprocessingConfig::getNucFilenameNoExtension(std::string_view nuc_name
-) const {
+std::filesystem::path PreprocessingConfig::getNucFilenameNoExtension(size_t sequence_idx) const {
    std::filesystem::path filename = input_directory;
-   filename /= fmt::format("{}{}", nucleotide_sequence_prefix, nuc_name);
+   filename /= fmt::format("{}{}", nucleotide_sequence_prefix, sequence_idx);
    return filename;
 }
```

Comment:
IMO it's not good to expose the sequences index to the outside. Users will have to provide sequence files using this index and thus it relies on the order in which you provide it in the JSON file.

As a user I would expect that those reference genomes yield the same result and that I have to provide the same input files:
```json
{
  "nucleotideSequences": [
    {
      "name": "first",
      "sequence": "T"
    },
    {
      "name": "second",
      "sequence": "T"
    }
  ],
  "genes": []
}
``` 

```json
{
  "nucleotideSequences": [
    {
      "name": "second",
      "sequence": "T"
    },
    {
      "name": "first",
      "sequence": "T"
    }
  ],
  "genes": []
}
``` 

If I understand the code correctly, that's currently not the case, because `nuc_0.fasta` would be the `first` sequence with the first reference genomes, but the `second` sequence with the second reference genomes.

Reply:
The thing is, that I wanted to allow all sequence names, that are provided by the user and work with duckdb. This means, that I would not want to create folder-names that contain these possibly weird symbols in all cases.

Why I thought referencing everything by index makes sense, is because we wanted to move the sequence definitions into the database_config anyways, where we would also just have a list of sequences, whose position would determine the index. As I wanted to not blow up the PR again, I first took the indexes from the reference_genomes